financial-templates-lib/clients/ExpiringMultiPartyClient.js

```diff
@@ -147,24 +149,23 @@ class ExpiringMultiPartyClient {
     this.expiredLiquidations = expiredLiquidations;
     this.disputedLiquidations = disputedLiquidations;
 
-    this.positions = this.sponsorAddresses.reduce(
-      (acc, address, i) =>
-        // Filter out empty positions.
-        positions[i].rawCollateral.toString() === "0"
-          ? acc
-          : /* eslint-disable indent */
-            acc.concat([
-              {
-                sponsor: address,
-                withdrawalRequestPassTimestamp: positions[i].withdrawalRequestPassTimestamp,
-                withdrawalRequestAmount: positions[i].withdrawalRequestAmount.toString(),
-                numTokens: positions[i].tokensOutstanding.toString(),
-                amountCollateral: collateral[i].toString(),
-                hasPendingWithdrawal: positions[i].withdrawalRequestPassTimestamp > 0
-              }
-            ]),
-      []
-    );
+    this.positions = this.sponsorAddresses.reduce((acc, address, i) => {
+      const rawCollateral = this.toBN(positions[i].rawCollateral.toString());
+      // Filter out empty positions.
+      return rawCollateral.isZero()
+        ? acc
+        : /* eslint-disable indent */
+          acc.concat([
+            {
+              sponsor: address,
+              withdrawalRequestPassTimestamp: positions[i].withdrawalRequestPassTimestamp,
+              withdrawalRequestAmount: positions[i].withdrawalRequestAmount.toString(),
+              numTokens: positions[i].tokensOutstanding.toString(),
+              amountCollateral: rawCollateral.mul(this.cumulativeFeeMultiplier).div(this.toBN(this.toWei("1"))),
+              hasPendingWithdrawal: positions[i].withdrawalRequestPassTimestamp > 0
+            }
+          ]);
+    }, []);
     this.lastUpdateTimestamp = await this.emp.methods.getCurrentTime().call();
```

Comment:
this contract query is made last no matter what to make sure that the `lastUpdateTimestamp` stays accurate to being greater than the last contract query

Reply:
I moved this call into the same `Promise.all` as `positions` an `getLiquidations`